src/TaskCreationSection/CategoryCreationSection/AddCategoryModal.jsx

```diff
@@ -76,67 +66,64 @@ export default function AddCategoryModal({ addCategory }) {
       });
       return;
     }
-    if (newCategory.name !== "" && isCategoryAlreadyAdded()) {
-      addCategory({
-        name: newCategory.name.toLowerCase(),
-        icon: newCategory.icon,
-        colorStyle: newCategory.colorStyle,
-        isAddedByUser: true,
-        uuid: uuid(),
-      });
+    const isCategoryValid =
+      newCategory.name &&
+      !categories.some(({ name }) => newCategory.name === name);
 
-      setNewCategory({ ...newCategory, name: "" });
-      showAlert({
-        title: "New category has been added!",
-        type: "success",
-        background: "bg-success",
-        isShowed: true,
-      });
-    } else {
+    if (!isCategoryValid) {
       setIsCorrectTyped(false);
       showAlert({
         title: "You can't create a category with this name, try again!",
         type: "error",
         background: "bg-error",
         isShowed: true,
       });
+      return;
     }
+
+    addCategory({ ...newCategory, uuid: uuid() });
+    showAlert({
+      title: "New category has been added!",
+      type: "success",
+      background: "bg-success",
+      isShowed: true,
+    });
   };
 
   const iconPicker = useMemo(
     () => (
       <IconPicker
-        newCategoryIcon={newCategory.icon}
-        setNewCategoryIcon={(icon) => setNewCategory({ ...newCategory, icon })}
         searchIcon={searchIcon}
         setSearchIcon={setSearchIcon}
+        newCategoryIcon={newCategory.icon}
+        setNewCategoryIcon={(icon) => setNewCategory({ ...newCategory, icon })}
       />
     ),
     [newCategory.icon]
   );
 
   const changeColorHandler = (color) => {
-    setColorPickerColor(color.hex);
     setNewCategory({
       ...newCategory,
       colorStyle: colorPickerColorHandler(color),
+      color: color.hex,
     });
   };
 
   const colorPicker = useMemo(
     () => (
       <CirclePicker
         className="self-center p-0 m-0"
-        color={colorPickerColor}
+        color={newCategory.color}
         colors={["#38bdf8", "#f87171", "#10b981", "#7e22ce", "#eab308"]}
         onChangeComplete={changeColorHandler}
       />
     ),
-    [colorPickerColor]
+    [newCategory.color]
   );
 
-  return (
-    <div>
+  function OpenModalButton() {
```

Comment:
very nice that you extracted smaller components but we should keep them outside of the main one like
```
const MainComponent = () => {
 // logic here
return <ComponentA propA={state.propA} />
}

const ComponentA = ({ propA }) => {}
```